logic/src/commonMain/kotlin/com/wire/kalium/logic/data/team/TeamRepository.kt

```diff
@@ -83,4 +91,43 @@ internal class TeamDataSource(
             teamsApi.deleteConversation(conversationId.value, teamId)
         }
     }
+
+    override suspend fun updateMemberRole(teamId: String, userId: String, permissionCode: Int?): Either<CoreFailure, Unit> {
+        return wrapStorageRequest {
+            val user = userMapper.fromTeamMemberToDaoModel(
+                teamId = TeamId(teamId),
+                nonQualifiedUserId = userId,
+                userDomain = selfUserId.domain,
+                permissionCode = permissionCode
+            )
+            userDAO.upsertTeamMembersTypes(listOf(user))
+        }
+    }
+
+    override suspend fun fetchTeamMember(teamId: String, userId: String): Either<CoreFailure, Unit> {
+        return wrapApiRequest {
+            teamsApi.getTeamMember(
+                teamId = teamId,
+                userId = userId,
+            )
+        }.flatMap {
+            wrapStorageRequest {
+                val user = userMapper.fromTeamMemberToDaoModel(
+                    teamId = TeamId(teamId),
+                    nonQualifiedUserId = userId,
+                    userDomain = selfUserId.domain,
+                    permissionCode = it.permissions?.own
+                )
+                userDAO.upsertTeamMembersTypes(listOf(user))
+            }
+        }
+    }
+
+    override suspend fun removeTeamMember(teamId: String, userId: String): Either<CoreFailure, Unit> {
+        TODO("Not yet implemented")
```

Comment:
Could we replace this for a logger, so we don't crash if we reach this point :D

Reply:
Added entire feature 😄